components/raftstore/src/store/worker/pd.rs

```diff
@@ -1114,6 +1114,9 @@ where
             .region_keys_read
             .observe(region_stat.read_keys as f64);
 
+        // TODO: this should only be done if enabled in config.
+        self.coprocessor_host
```

Comment:
I think all regions will be considered as `active_regions` since they periodically report their heartbeat.

and we already record the region_stat of active regions in `region_peers`, we can collect the active topN region like https://github.com/tikv/tikv/blob/66847e9c5ac701e73da1366938a7e2306cacc671/components/raftstore/src/store/worker/pd.rs#L1143-L1169
then we don't need to observe the heartbeat event.

Reply:
Maybe it's more efficient to get top N in pd.rs. However I don't know how to pass this information to RegionCollector in region_info_accessor.rs. The final decision to evict or load a region into cache will depend on other factors, such as region label (e.g. manual hint) and mvcc versions or tombstones, etc.

Reply:
I am leaning towards keeping things as is for now, but then perhaps integrating with StatsMonitor.

Reply:
Spoke offline with @tonyxuqqi  we decided to stick with my retention based approach to avoid complicating the stats monitor. Over all the N here (maximum possible number of regions) is rather small, what we can do is just retain only the top regions for now.